pkgs/top-level/all-packages.nix

```diff
@@ -13765,8 +13765,21 @@ with pkgs;
 
   pipenv = callPackage ../development/tools/pipenv {};
 
-  pipewire = callPackage ../development/libraries/pipewire {};
+  pipewire =
+    let
+      # Break a dependency cycle
+      customSDL2 = SDL2.override {
+        pipewireSupport = false;
+      };
+    in callPackage ../development/libraries/pipewire {
+      SDL2 = customSDL2;
+      ffmpeg = ffmpeg.override {
+        SDL2 = customSDL2;
+      };
```

Comment:
Do we _need_ SDL support in regular ffmpeg? That sounds more like a feature for ffmpeg-full to me.

Again, closure size concerns.

Reply:
Hmm, It's here to avoid causing infinite recursion which happens without it, which makes me think that `ffmpeg` in pipewire requires SDL2.

I'm not exactly sure why does this require SDL2, though.

Reply:
I know why these are here but I'd rather avoid doing it entirely.

Reply:
Hmm, I'd have to check why does it require SDL2, then.

Reply:
SDL2 is used at least in libavdevice for the SDL and OpenGL output devices and in ffplay. Probably fine to remove from the non-full package to reduce closure size.

Reply:
Losing ffplay wouldn't be acceptable IMO.

Luckily, if SDL doesn't need pipewire, we don't need to worry about this.

Reply:
You mean if pipewire doesn't need SDL?

Reply:
Yes, That way around \^\^.

Reply:
Should we just keep passing `customSDL2` to ffmpeg required in pipewire, then?

Reply:
Hmm, pipewire only actually needs ffmpeg for the libavcodec plugin. Is there a way to pass in an "ffmpeg" with only libavcodec and headers?

Reply:
Why though? Can't we just pass regular SDL to ffmpeg once that no longer depends on pipewire?

Reply:
ffmpeg requires SDL2 for ffplay, SDL2 requires pipewire for audio output, pipewire requires ffmpeg for libavcodec plugin

Reply:
Is it possible to disable/strip certain buildInputs from one output only (dev)?